src/inmanta/server/services/databaseservice.py

```diff
@@ -90,6 +90,12 @@ async def connect_database(self) -> None:
         )
         LOGGER.info("Connected to PostgreSQL database %s on %s:%d", opt.db_name.get(), database_host, database_port)
 
+        # Check if JIT is enabled
+        async with self._pool.acquire() as connection:
+            jit_enabled = await connection.fetchval("SHOW jit;")
```

Comment:
You will need to rely on `SELECT pg_jit_available();` instead of `SHOW jit;`. For the jit to work, two things are required: The jit compiler needs to be installed and the jit config option needs to be enabled. The `SHOW jit;` command only shows the value of the config option, while the `SELECT pg_jit_available();` returns only true if both requirements are met. Using the latter command will ensure we only raise the warning when the jit is actually enabled. This is especially relevant because the default value of the jit config option it `on`.